test/unit_tests/test_metadata_table.py

```diff
@@ -0,0 +1,141 @@
+       filename = os.path.join(sample_files_dir, "test_bad_type_name.meta")
+
+       #Exercise
+       with self.assertRaises(Exception) as context:
+           MetadataTable.parse_metadata_file(filename, known_ddts, logger)
+
+       #Verify
+       #print("The exception is", context.exception)
+       self.assertTrue('Invalid metadata table type, \'banana' in str(context.exception))
+
+   def test_double_header(self):
+       """Test that a duplicate header returns expected error"""
+       known_ddts = list()
+       logger = None
+       filename = os.path.join(sample_files_dir, "double_header.meta")
+
+       with self.assertRaises(Exception) as context:
+           MetadataTable.parse_metadata_file(filename, known_ddts, logger)
+
+       #print("The exception is", context.exception)
+       self.assertTrue('Duplicate metadata header, test_host' in str(context.exception))
+
+   def test_bad_dimension(self):
+       """Test that `dimension = banana` returns expected error"""
+       known_ddts = list()
+       logger = None
+       filename = os.path.join(sample_files_dir, "test_bad_dimension.meta")
+
+       with self.assertRaises(Exception) as context:
+           MetadataTable.parse_metadata_file(filename, known_ddts, logger)
+
+       #print("The exception is", context.exception)
+       self.assertTrue('Invalid \'dimensions\' property value, \'' in str(context.exception))
+
+   def test_duplicate_variable(self):
+       """Test that a duplicate variable returns expected error"""
+       known_ddts = list()
+       logger = None
+       filename = os.path.join(sample_files_dir, "test_duplicate_variable.meta")
+
+       with self.assertRaises(Exception) as context:
+           MetadataTable.parse_metadata_file(filename, known_ddts, logger)
+
+       #print("The exception is", context.exception)
+       self.assertTrue('Invalid (duplicate) standard name in temp_calc_adjust_run, defined at ' in str(context.exception))
+
+   def test_invalid_intent(self):
+       """Test that an invalid intent returns expected error"""
+       known_ddts = list()
+       logger = None
+       filename = os.path.join(sample_files_dir, "test_invalid_intent.meta")
+
+       with self.assertRaises(Exception) as context:
+           MetadataTable.parse_metadata_file(filename, known_ddts, logger)
+
+       #print("The exception is", context.exception)
+       self.assertTrue('Invalid \'intent\' property value, \'banana\', at ' in str(context.exception))
+
+   def test_missing_intent(self):
+       """Test that a missing intent returns expected error"""
+       known_ddts = list()
+       logger = None
+       filename = os.path.join(sample_files_dir, "test_missing_intent.meta")
+
+       with self.assertRaises(Exception) as context:
+           MetadataTable.parse_metadata_file(filename, known_ddts, logger)
+
+       print("The exception is", context.exception)
+       #self.assertTrue('Missing \'intent\' for variable \'timestep\', at ' in str(context.exception))
```

Comment:
Put this back to generating the error (i.e., comment line 137 and uncomment line 138).